app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala

```diff
@@ -365,6 +384,24 @@ class SubsidyController @Inject()(
     )
   }
 
+  def getChangeSubsidyClaim(transactionId: String): Action[AnyContent] = escAuthentication.async { implicit request =>
+    implicit val eori: EORI = request.eoriNumber
+    for {
+      undertaking <- store.get[Undertaking]
+      reference = undertaking.getOrElse(throw new IllegalStateException("")).reference.getOrElse(throw new IllegalStateException(""))
```

Comment:
Could you do something like

```
reference = undertaking.flatMap(_.reference).getOrElse(...)
```

instead, so you only need to define the `None` case once?